test/bats/azure.bats

```diff
@@ -125,13 +126,70 @@ setup() {
   [[ "${result//$'\r'}" -eq "${SECRET_VALUE}" ]]
 
   result=$(kubectl get secret foosecret -o json | jq '.metadata.ownerReferences | length')
-  [[ "$result" -eq "2" ]]
+  [[ "$result" == "2" ]]
 }
 
 @test "Sync with K8s secrets - delete deployment, check secret deleted" {
   run kubectl delete -f $BATS_TESTS_DIR/nginx-deployment-synck8s-azure.yaml
   assert_success
+
+  run kubectl delete -f $BATS_TESTS_DIR/azure_synck8s_v1alpha1_secretproviderclass.yaml
+  assert_success
+
   sleep 20
   result=$(kubectl get secret | grep foosecret | wc -l)
   [[ "$result" -eq "0" ]]
 }
+
+@test "Test Namespaced scope SecretProviderClass - create deployment" {
+  run kubectl create ns test-ns
+  assert_success
+
+  run kubectl create secret generic secrets-store-creds --from-literal clientid=${AZURE_CLIENT_ID} --from-literal clientsecret=${AZURE_CLIENT_SECRET} -n test-ns
+  assert_success
+
+  envsubst < $BATS_TESTS_DIR/azure_v1alpha1_secretproviderclass_ns.yaml | kubectl apply -f -
+
+  cmd="kubectl wait --for condition=established --timeout=60s crd/secretproviderclasses.secrets-store.csi.x-k8s.io"
+  wait_for_process $WAIT_TIME $SLEEP_TIME "$cmd"
+
+  cmd="kubectl get secretproviderclasses.secrets-store.csi.x-k8s.io/azure-sync -o yaml | grep azure"
```

Comment:
why are we checking this in default ns? seems it's deleted on line 136

Reply:
`azure_v1alpha1_secretproviderclass_ns.yaml` creates 2 spc with the same name, one in `default` namespace and other in `test-ns`. This test ensures the spc in the same namespace as pod is matched and not the one in default namespace. The default namespace spc has an invalid provider - https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/264/files#diff-b09039a0f14308e43877bb2112de35f1R6 which is a good sign to ensure the correct SPC is always matched

Reply:
ah ok. i missed that.